### DISCUSSION TITLE
[Proposal] Add post-update options for Hugo modules

### DISCUSSION THREAD

**Discussion by razonyang** [renovatebot/renovate#21017] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Hi, the [Hugo modules](https://gohugo.io/hugo-modules/) base on the Go modules, it works fine without `postUpdateOptions`.

I would like to see there is a post-update option like `gomodTidy` for Hugo modules, the `gomodTidy` will remove all dependencies from `go.mod` and `go.sum` ([example](https://github.com/razonyang/hb-theme/pull/18/files)), since there is no any Go source code in Hugo modules, and Hugo uses the command `hugo mod tidy` instead of `go mod tidy` to add missing and remove unused modules.

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "matchManagers": ["gomod"],
      "matchDepTypes": ["indirect"],
      "enabled": true
    }
  ],
  "postUpdateOptions": [
    "gomodTidy"
  ]
}
```

**Comment by HonkingGoose** [comment]
I found these relevant docs from the official Hugo project:

- [`hugo mod` command](https://gohugo.io/commands/hugo_mod/) 
- [`hugo mod tidy` command](https://gohugo.io/commands/hugo_mod_tidy/)
- [Hugo Modules Overview](https://gohugo.io/hugo-modules/)
- [Configure Modules](https://gohugo.io/hugo-modules/configuration/)
- [Use Hugo Modules](https://gohugo.io/hugo-modules/use-modules/)

**Reply by HonkingGoose** [reply]
@rarkins can you take a look at this idea, and tell me if you want an issue for it?

**Comment by rarkins** [comment]
I don't really understand the question. Do Hugo modules already mostly work today because they use `go.mod`/`go.sum`?

**Reply by razonyang** [reply]
Thanks for spending time on this, Renvatebot works nice with Hugo modules, since Hugo modules base on Go modules(go.mod, go.sum), but there is a different that `go mod tidy` will remove all dependencies from Hugo module's `go.sum`, because there is no Go source code inside Hugo modules, Hugo uses [`hugo mod tidy`](https://gohugo.io/commands/hugo_mod_tidy/) instead `go mod tidy` to remove unused modules.

**Reply by rarkins** [reply]
And what happens if there's _no_ go mod tidy run? (Which I think is the default?)

I think we could add a post update option for Hugo's mod tidy but it would require adding Hugo installation support to https://github.com/containerbase/base first

**Reply by razonyang** [reply]
> And what happens if there's no go mod tidy run? (Which I think is the default?)

Anything works as expected.
